driver/compile.go

```diff
@@ -4,21 +4,73 @@ import (
 	"context"
 
 	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
 	"github.com/brimsec/zq/zng/resolver"
 	"go.uber.org/zap"
 )
 
-func Compile(program ast.Proc, input proc.Proc) (*proc.MuxOutput, error) {
-	ctx := &proc.Context{
-		Context:     context.Background(),
+// Compile takes an AST, an input reader, and configuration parameters,
+// and compiles it into a runnable flowgraph, returning a
+// proc.MuxOutput that which brings together all of the flowgraphs
+// tails, and is ready to be Pull()'d from.
+func Compile(ctx context.Context, program ast.Proc, reader zbuf.Reader, reverse bool, span nano.Span, logger *zap.Logger) (*proc.MuxOutput, error) {
```

Comment:
This function signature is a bit big... it reflects the fact that a bunch of stuff goes into wiring up a flowgraph. 

We could also pass in the `proc.Context` instead of the last three parameters, but that would then mean more prepwork at the call sites.